vm/src/dictdatatype.rs

```diff
@@ -557,10 +557,13 @@ impl<T: Clone> Dict<T> {
         let removed = if entry_index == inner.used {
             inner.entries.pop().unwrap()
         } else {
-            let last_index = inner.entries.last().unwrap().index;
-            let removed = inner.entries.swap_remove(entry_index);
```

Comment:
oh... I believed it is not swap_remove and we were doing it as how CPython does.